Src/BootCamp.Chapter/BalanceStats.cs

```diff
@@ -1,37 +1,224 @@
     {
         /// <summary>
         /// Return name and balance(current) of person who had the biggest historic balance.
-        /// </summary>
-        public static string FindHighestBalanceEver(string[] peopleAndBalances)
+        /// </summary>       
+        public static string FindHighestBalanceEver(string[] peoplesBalances)
         {
-            return "";
+            if (IsNullOrEmpty(peoplesBalances))
+
+                return "N/A.";
+
+            var (peopleNames, cash) = CalculatePB(peoplesBalances, "max");
+
+            return $"{peopleNames} had the most money ever. {FormatCash(cash)}.";
         }
 
         /// <summary>
         /// Return name and loss of a person with a biggest loss (balance change negative).
         /// </summary>
-        public static string FindPersonWithBiggestLoss(string[] peopleAndBalances)
+        public static string FindPersonWithBiggestLoss(string[] peoplesBalances)
         {
-            return "";
+            if (IsNullOrEmpty(peoplesBalances))
+
+                return "N/A.";
+
+            var (peopleNames, cash) = CalculatePB(peoplesBalances, "min");
+
+            if (cash >= 0)
+
+                return "N/A.";
+
+            return $"{peopleNames} lost the most money. {FormatCash(cash)}.";
         }
 
         /// <summary>
         /// Return name and current money of the richest person.
         /// </summary>
-        public static string FindRichestPerson(string[] peopleAndBalances)
+
+        public static string FindRichestPerson(string[] peoplesBalances)
         {
-            return "";
+            if (IsNullOrEmpty(peoplesBalances))
+
+                return "N/A.";
+
+            var (peopleNames, cash) = CalculatePB(peoplesBalances, "rich");
+            var subjects = PeoplePlus(peopleNames);
+            var word = new string[]
+            {
+                (subjects) ? "are" : "is",
+                (subjects) ? "people" : "person"
+            };
+            return $"{peopleNames} {word[0]} the richest {word[1]}. {FormatCash(cash)}.";
         }
 
         /// <summary>
         /// Return name and current money of the most poor person.
         /// </summary>
-        public static string FindMostPoorPerson(string[] peopleAndBalances)
+        public static string FindMostPoorPerson(string[] peoplesBalances)
         {
-            return "";
+            if (IsNullOrEmpty(peoplesBalances))
+
+                return "N/A.";
+
+            var (peopleNames, cash) = CalculatePB(peoplesBalances, "poor");
+            var subjects = PeoplePlus(peopleNames);
+            var word = new string[]
+            {
+                (subjects) ? "have" : "has",
+            };
+
+            return $"{peopleNames} {word[0]} the least money. {FormatCash(cash)}.";
+        }
+
+        private static (StringBuilder peopleNames, float cash) CalculatePB(string[] peoplesBalances, string attribution)
+        {
+            var validation = new string[]
```

Comment:
validation is very vague, please rename better.

Reply:
Naming is hard!